pkg/controller/daemonset/daemonset_controller.go

```diff
@@ -923,10 +923,19 @@ func (dsc *ReconcileDaemonSet) getNodesToDaemonPods(ds *appsv1alpha1.DaemonSet)
 		}
 		node := &corev1.Node{}
 		err = dsc.client.Get(context.TODO(), types.NamespacedName{Name: nodeName}, node)
-		if err == nil && CanNodeBeDeployed(node, ds) {
+		if err != nil {
+			if errors.IsNotFound(err) {
+				// Unable to find the target node in the cluster, which means it has been drained from cluster,
```

Comment:
Correct me if I am wrong. If the node object is deleted from the apiserver, I think the default gc controller will remove the Pods that are associated to the deleted node. Is it true?

My understanding of the reported problem is that for the drained node, the "CanNodeBeDeployed" method returns false hence the Pods are not added to the map.

Reply:
> If the node object is deleted from the apiserver, I think the default gc controller will remove the Pods that are associated to the deleted node.

Right, If a pod has been scheduled to the node, it will be removed. However, if a pod has been created and added nodeAffinity by daemonset-controller but has not been scheduled to the node, it will stuck in pending state and will not be added to the map, also.

Reply:
Now I understand the problem when looking at the util.GetTargetNodeName().